spinta/backends/postgresql/types/geometry/encode.py

```diff
@@ -40,8 +44,17 @@ def prepare_dtype_for_response(
     action: Action,
     select: dict = None,
 ):
-    shape = to_shape(value)
+    if value.srid not in [4326, -1]:
+        wgs_crs = CRS('EPSG:4326')
+        srid_crs = CRS(f'EPSG:{value.srid}')
+        project = Transformer.from_crs(crs_from=srid_crs, crs_to=wgs_crs, always_xy=True).transform
+        shape = transform(project, to_shape(value))
+    else:
+        shape = to_shape(value)
+
     point = shape.centroid
-    params = urlencode({'mlat': point.x, 'mlon': point.y})
-    link = f'https://www.openstreetmap.org/?{params}#map=19/{point.x}/{point.y}'
-    return Cell(shape.wkt, link=link)
+    x, y = point.x, point.y
+    params = urlencode({'mlat': x, 'mlon': y})
+    link = f'https://www.openstreetmap.org/?{params}#map=19/{x}/{y}'
+    shape_type = shape.type == 'Point' and shape.wkt or shape.type.upper()
+    return Cell(shape_type, link=link)
```

Comment:
```suggestion
    if isinstance(shape, Point):
        value = shape.wkt
    else:
        # Shorten possibly long WKT values by reducing it to shape type.
        value = shape.type.upper()
    return Cell(value, link=link)
```

Where `Point` comes from `from shapely.geometry import Point`.

Consider this suggestion, because:

- variable name `shape_type` might be shape_type or wkt value, so better name things correctly,
- avoid relying on raw values like `'Point'` in this case, better use constants and class names, they are more reliable and Python can check if a name `Point` exists.
- avoid cryptic code like `condition and then or else`, just use `if condition: then; else: else` or at least `then if condition else else`.